addons/mrp_subcontracting/views/stock_move_views.xml

```diff
@@ -18,7 +18,7 @@
                 <field name="state" invisible="1"/>
                 <!-- Don't put move_id here to avoid that the framework send falsy move_id -->
                 <field name="id" invisible="1"/>
-                <field name="product_id" required="1"/>
+                <field name="product_id" required="1" domain="[('id', 'in', context.get('bom_product_ids'))] if context.get('is_subcontracting_portal') else []"/>
```

Comment:
I would rather do a field on product.product computed on the context and the active user. It would avoid to play with the context